api/resources_portal/views/user.py

```diff
@@ -52,21 +41,41 @@ class Meta:
         extra_kwargs = {"password": {"write_only": True}}
 
 
-class UserCreateViewSet(mixins.CreateModelMixin, viewsets.GenericViewSet):
-    """
-    Creates user accounts
-    """
+class IsUserOrAdmin(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return request.user == obj or request.user.is_superuser
+
 
+class UserViewSet(viewsets.ModelViewSet):
     queryset = User.objects.all()
-    serializer_class = CreateUserSerializer
-    permission_classes = (AllowAny,)
+    serializer_class = UserSerializer
 
     @transaction.atomic
     def create(self, request, *args, **kwargs):
-        response = super(UserCreateViewSet, self).create(request, args, kwargs)
+        response = super(UserViewSet, self).create(request, args, kwargs)
 
         # Every user should have their own organization.
         user = User.objects.get(id=response.data["id"])
         Organization.objects.create(owner=user)
 
         return response
+
+    def get_serializer_class(self):
+        if self.action == "create":
+            return CreateUserSerializer
+
+        return UserSerializer
+
+    def get_permissions(self):
+        if self.action == "list" or self.action == "destroy":
+            permission_classes = [IsAdminUser]
+        elif (
+            self.action == "retrieve" or self.action == "update" or self.action == "partial_update"
+        ):
+            permission_classes = [IsAuthenticated, IsUserOrAdmin]
+        elif self.action == "create":
+            permission_classes = [AllowAny]
+        else:
+            permission_classes = [IsAuthenticated]
```

Comment:
Ok so this just reiterates the comments I left in the tests, but I think `retrieve` and `list` should be moved down to the else, `destroy` should be moved into the second `elif` (so they have IsUserOrAdmin), and then the first `if` can be removed.